Modules/pyexpat.c

```diff
@@ -110,11 +110,15 @@ struct HandlerInfo {
 
 static struct HandlerInfo handler_info[64];
 
-#define CALL_XML_HANDLER_SETTER(HANDLER_INFO, XML_PARSER, XML_HANDLER)      \
-    do {                                                                    \
-        xmlhandlersetter setter = (xmlhandlersetter)(HANDLER_INFO).setter;  \
-        setter((XML_PARSER), (XML_HANDLER));                                \
-    } while (0)
+// gh-111178: Use _Py_NO_SANITIZE_UNDEFINED, rather than using the exact
+// handler API for each handler.
+static inline void _Py_NO_SANITIZE_UNDEFINED
+CALL_XML_HANDLER_SETTER(const struct HandlerInfo *handler_info,
+                        XML_Parser xml_parser, xmlhandler xml_handler)
+{
+    xmlhandlersetter setter = (xmlhandlersetter)(handler_info->setter);
```

Comment:
```suggestion
    xmlhandlersetter setter = (xmlhandlersetter)handler_info->setter;
```

Parenthese are no more needed I think.